guide/writing-code/uppercaser.md

```diff
@@ -218,7 +218,7 @@ _main:
   ; ("./uppercaser-mac.asm")
   mov rbx, qword [rcx + 8]
 
-  ; Push the memory address of our command-line arg onto the stack twice.
+  ; Push the value of our command-line arg onto the stack twice.
```

Comment:
Ooooh nice catch!! I think we actually want to meet somewhere in the middle for this.

You're absolutely correct that `[]` gets us a value out of memory. However, strings are tricky - they're arrays themselves, so an array of strings is actually an array of arrays. So when we follow that first memory address, we find ourselves at... yet another memory address. It points to the first character in our string.

So, how about something like:

```suggestion
  ; Push the memory address of the first character of our second command line arg onto the stack twice.
```

<br />

Does this wording make sense? 🤔

Reply:
Ha that's tricky.  Now that I know, I think the original sentence makes sense (could be stated either way).  Your suggestion clears it up however since the label is called in a loop, it's not always the second command line arg, right?  I added a suggestion as well in case others have the same confusion as I did..